src/main/java/com/thebois/abstractions/IPositionFinder.java

```diff
@@ -0,0 +1,30 @@
+package com.thebois.abstractions;
+
+import com.thebois.models.Position;
+
+/**
+ * Allows for locating certain types of positions in the world.
+ */
+public interface IPositionFinder {
+
+    /**
+     * Creates a list of tiles that are not occupied by a structure or resource.
+     *
+     * @param count The amount of empty positions that needs to be found.
+     *
+     * @return List of empty positions.
+     */
+    Iterable<Position> findEmptyPositions(int count);
+
+    /**
+     * Creates a list of tiles that are not occupied by a structure or resource.
```

Comment:
style

Maybe a bit too implementation-specific, since we might add more things to tiles that make them "occupied", it's very hard to keep comments like these up-to-date.